lib/bundler/installer/parallel_installer.rb

```diff
@@ -131,6 +135,12 @@ def handle_error
       raise Bundler::InstallError, errors.map(&:to_s).join("\n\n")
     end
 
+    def check_for_corrupt_lockfile
+      return unless @specs.any? {|s| s.corrupt_lockfile?(@specs) }
+      Bundler.ui.warn "Using 1 thread to install specs instead of #{@size} due to lockfile corruption" unless @size == 1
```

Comment:
That will also mean returning the array of missing names from the method above, I think?